dspace-api/src/test/java/org/dspace/discovery/DiscoveryTest.java

```diff
@@ -0,0 +1,124 @@
+                                                                     .getServiceByName(IndexingService.class.getName(),
+                                                                                       IndexingService.class);
+    protected SearchService searchService = SearchUtils.getSearchService();
+
+
+    WorkspaceItem leftIs;
+    WorkspaceItem rightIs;
+
+    /**
+     * This method will be run before every test as per @Before. It will
+     * initialize resources required for the tests.
+     *
+     * Other methods can be annotated with @Before here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @Before
+    @Override
+    public void init() {
+        super.init();
+        try {
+            context.turnOffAuthorisationSystem();
+            Community community = communityService.create(null, context);
+
+            Collection col = collectionService.create(context, community);
+            leftIs = workspaceItemService.create(context, col, false);
+            rightIs = workspaceItemService.create(context, col, false);
+
+            itemService.addMetadata(context, leftIs.getItem(), "dc", "description", "abstract", null, "headache");
+            itemService.addMetadata(context, rightIs.getItem(), "dc", "description", "abstract", null, "headache");
+
+            workspaceItemService.update(context, leftIs);
+            workspaceItemService.update(context, rightIs);
+
+            indexingService.commit();
+
+            context.restoreAuthSystemState();
+            context.dispatchEvents();
+        } catch (AuthorizeException ex) {
+            log.error("Authorization Error in init", ex);
+            fail("Authorization Error in init: " + ex.getMessage());
+        } catch (SQLException ex) {
+            log.error("SQL Error in init", ex);
+            fail("SQL Error in init: " + ex.getMessage());
+        } catch (Exception ex) {
+            log.error("Error in init", ex);
+            fail("Error in init: " + ex.getMessage());
+        }
+    }
+
+    /**
+     * This method will be run after every test as per @After. It will
+     * clean resources initialized by the @Before methods.
+     *
+     * Other methods can be annotated with @After here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @After
+    @Override
+    public void destroy() {
+        context.abort();
+        super.destroy();
+    }
+
+    @Test
+    public void deleteWorkspaceItemSolrRecordAfterDeletionFromDbTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        DiscoverQuery discoverQuery = new DiscoverQuery();
+        discoverQuery.setQuery("*:*");
+        discoverQuery.addFilterQueries("search.resourceid:" + leftIs.getID());
+        DiscoverResult discoverResult = searchService.search(context, discoverQuery);
+        List<IndexableObject> indexableObjects = discoverResult.getIndexableObjects();
+        assertTrue(indexableObjects.size() == 1);
```

Comment:
@KevinVdV : here's the main line where Travis CI is throwing the error.  Unfortunately, all we know is this ends up being "false".  Could we change this assert to provide us more information?  It probably should instead say something like:
`assertEquals(1, indexableObjects.size());`

That would at least let us know if IndexableObjects is *zero* initially (in which case Solr isn't starting up correctly) or maybe it's some number >1 (in which case either your query is too broad, or Solr isn't cleaning up data between tests & we have data leakage across tests again).